### PR TITLE
sor10874/readmes/25-30

### PR DESCRIPTION
**Pull request by sorenoid**
this PR covers

IdentifyKMLFeatures
IdentifyLayers (java and kotlin)
IntegratedWindowsAuthentication
LineOfSightGeoElement
LineOfSightLocation

LineOfSightGeoElement "How it works" had to be rewritten substantially. I left method names in "Relevant API" sections as written in the common samples but I did remove parentheses like other samples do, e.g. IdentifyKMLFeatures

I also removed the "offline data" section for the "line of sight geoelement" README because the files are shipped with the app as assets.

### CODE REVIEW

File: java/identify-kml-features/README.md [hunk 1]
```diff
@@ -18,28 +18,30 @@ Note: the KML layer used in this sample contains a screen overlay. The screen ov
 
 1. Create an `OnTouchListener` on the `MapView`.
 2. On tap:
-  * Dismiss the callout.
+  * Dismiss the `Callout`, if one is showing.
   * Call `identifyLayersAsync(...)` passing in the `KmlLayer`, screen point and tolerance.
```
Comment (hud10837) [inline java/identify-kml-features/README.md]: Layer should be singular in the method call, and it probably makes sense to include the class name as well.
```suggestion
  * Call `MapView.identifyLayerAsync(...)` passing in the `KmlLayer`, screen point and tolerance.
```

File: java/identify-kml-features/README.md [hunk 2]
```diff
@@ -18,28 +18,30 @@ Note: the KML layer used in this sample contains a screen overlay. The screen ov
...
-  * Use the placemark to position and populate the content of a callout.
+  * Await the result of the identify and then get the `KmlPlacemark` from the result.
+  * Create a callout at the calculated map point and populate the callout content with text from the placemark's `BalloonContent`. NOTE: KML supports defining HTML for balloon content and may need to be converted from HTML to text.
   * Show the callout.
   
 Note: There are several types of KML features. This sample only identifies features of type `KmlPlacemark`.
 
 ## Relevant API
 
+* GeoView.identifyLayerAsync
```
Comment (hud10837) [inline java/identify-kml-features/README.md]: API shouldn't include method names
```suggestion
* GeoView
```

File: java/identify-kml-features/README.md [hunk 3]
```diff
@@ -18,28 +18,30 @@ Note: the KML layer used in this sample contains a screen overlay. The screen ov
...
-#### Tags
-Search and Query
-Weather
-NOAA
-NWS
-KML
-KMZ
-OGC
-Keyhole
+* KmlPlacemark.getBalloonContent
```
Comment (hud10837) [inline java/identify-kml-features/README.md]:
```suggestion
```

File: java/identify-layers/README.md [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
+![Image of identify layers](identify-layers.png)
+
+## Use case
+
+IdentifyLayers allows users to tap on a map, returning features at that location across multiple layers. Because some layer types have sublayers, the sample recursively counts results for sublayers within each layer.
 
 ## How to use the sample
-Tap on map to get features at that location. The features are returned from different layers in the map.
+
+Tap to identify features. An alert will show all layers with features under the cursor, as well as a layer count.
```
Comment (hud10837) [inline java/identify-layers/README.md]: Sample does not display a layer count, but does display the number of features within each layer.

File: java/identify-layers/README.md [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
 
 ## How it works
-`MapView` has a `identifyLayersAsync(screenLocation, tolerance, returnPopupsOnly, maximumResults)` method that is used in the sample. The method takes a screen location, tolerance, boolean for returning (a pop-up/pop-up and geo-element), and maximum results per layer, which results in a `ListenableFuture<List<IdentifyLayerResult>>`.
-	
-Layer name and a count of identified features held by each `IdentifyLayerResult` in the `ListenableFuture<List<...>>` is then taken and written out to a String. Finally, the resulting String is displayed in an Android AlertDialog.
+
+1. The tapped position is passed to `MapView.IdentifyLayersAsync`
```
Comment (hud10837) [inline java/identify-layers/README.md]: lower case method name
```suggestion
1. The tapped position is passed to `MapView.identifyLayersAsync`
```

File: java/identify-layers/README.md [hunk 6]
```diff
@@ -1,21 +1,34 @@
...
-* ArcGISMapImageLayer
-* FeatureLayer
-* FeatureTable
-* MapView
 
-#### Tags
-Search and Query
\ No newline at end of file
+* IdentifyLayerResult
+* IdentifyLayerResult.LayerContent.Name
```
Comment (hud10837) [inline java/identify-layers/README.md]: Remove method names in API section
```suggestion
```

File: java/identify-layers/README.md [hunk 7]
```diff
@@ -1,21 +1,34 @@
...
+* IdentifyLayerResult.SublayerResults
```
Comment (hud10837) [inline java/identify-layers/README.md]: ⬆️ 
```suggestion
```

File: java/identify-layers/README.md [hunk 8]
```diff
@@ -1,21 +1,34 @@
...
+* MapView.IdentifyLayersAsync
```
Comment (hud10837) [inline java/identify-layers/README.md]: ⏫ 
```suggestion
* MapView
```

File: java/identify-layers/README.md [hunk 9]
```diff
@@ -1,21 +1,34 @@
...
+
+## Additional information
+
+The GeoView supports two methods of identify: `identifyLayer()`, which identifies features within a specific layer and `identifyLayers()`, which identifies features for all layers in the current view.
```
Comment (hud10837) [inline java/identify-layers/README.md]: it looks like both of these methods are only available in async form, but I'm not 100% sure. Check here https://developers.arcgis.com/android/latest/api-reference/reference/com/esri/arcgisruntime/mapping/view/GeoView.html#identifyLayerAsync(com.esri.arcgisruntime.layers.Layer,android.graphics.Point,double,boolean,int)

```suggestion
The GeoView supports two methods of identify: `identifyLayerAsync()`, which identifies features within a specific layer and `identifyLayersAsync()`, which identifies features for all layers in the current view.
```

File: java/integrated-windows-authentication/README.md [hunk 10]
```diff
@@ -1,32 +1,47 @@
-# Integrated Windows Authentication
+# Integrated windows authentication
```
Comment (hud10837) [inline java/integrated-windows-authentication/README.md]: I can't really test this one unfortunately, and some of the stuff here is running behind the scenes, but I trust the API is all good!

File: java/line-of-sight-geoelement/README.md [hunk 11]
```diff
@@ -1,28 +1,34 @@
...
-![Line of Sight GeoElement App](line-of-sight-geoelement.png)
+A `GeoElementLineOfSight` is therefore useful in cases where visibility between two `GeoElement`s requires monitoring over a period of time in a partially obstructed field of view
+(such as buildings in a city).
 
 ## How to use the sample
 
-A line of sight will display between a point on the Empire State Building (observer) and a taxi (target). The taxi will drive around a block and the line of sight should automatically update. The taxi will be highlighted when it is visibile. You can change the observer height with the slider to see how it affects the target's visibility.
+A line of sight will display between a point on the Empire State Building (observer) and a taxi (target).
+The taxi will drive around a block and the line of sight should automatically update.
+The taxi will be highlighted when it is visible. You can change the observer height with the slider to see how it affects the target's visibility.
```
Comment (hud10837) [inline java/line-of-sight-geoelement/README.md]: It looks like the taxi isn't just highlighted, instead it has a flashing highlight? And the main thing is that the line of sight is red for the portion that is obscured.

File: java/line-of-sight-geoelement/README.md [hunk 12]
```diff
@@ -1,28 +1,34 @@
...
-1. To get the target visibility when it changes, add a `TargetVisibilityChangedListener` to the line of sight. The changed event will give the `TargetVisibility`.
+1. Instantiate an `AnalysisOverlay` and add it to the `SceneView`'s analysis overlays collection.
+2. Instantiate a `GeoElementLineOfSight`, passing in observer and target `GeoElement`s (features or graphics). Add the line of sight to the analysis overlay's analyses collection.
+3. To get the target visibility when it changes, react to the target visibility changing on the `GeoElementLineOfSight` instance.
 
 ## Relevant API
 
 * AnalysisOverlay
 * GeoElementLineOfSight
 * LineOfSight.TargetVisibility
```
Comment (hud10837) [inline java/line-of-sight-geoelement/README.md]:
```suggestion
* LineOfSight
```

File: java/location-line-of-sight/README.md [hunk 13]
```diff
@@ -1,25 +1,29 @@
-# Location Line of Sight
-Perform a line of sight analysis between two points in a SceneView.
+# Line of sight location
```
Comment (hud10837) [inline java/location-line-of-sight/README.md]: This one is called Location line of sight in the sample viewer

File: kotlin/identify-layers/README.md [hunk 14]
```diff
@@ -20,15 +20,15 @@ Tap to identify features. An alert will show all layers with features under the
 
 ## Relevant API
 
-* `IdentifyLayerResult`
-* `IdentifyLayerResult.LayerContent.Name`
-* `IdentifyLayerResult.SublayerResults`
-* `MapView.IdentifyLayersAsync`
+* IdentifyLayerResult
+* IdentifyLayerResult.LayerContent.Name
```
Comment (hud10837) [inline kotlin/identify-layers/README.md]:
```suggestion
```

File: kotlin/identify-layers/README.md [hunk 15]
```diff
@@ -20,15 +20,15 @@ Tap to identify features. An alert will show all layers with features under the
...
+* IdentifyLayerResult.SublayerResults
```
Comment (hud10837) [inline kotlin/identify-layers/README.md]:
```suggestion
```

File: kotlin/identify-layers/README.md [hunk 16]
```diff
@@ -20,15 +20,15 @@ Tap to identify features. An alert will show all layers with features under the
...
+* MapView.IdentifyLayersAsync
```
Comment (hud10837) [inline kotlin/identify-layers/README.md]:
```suggestion
* MapView
```

File: kotlin/identify-layers/README.md [hunk 17]
```diff
@@ -20,15 +20,15 @@ Tap to identify features. An alert will show all layers with features under the
...
 
 ## Additional information
 
 The GeoView supports two methods of identify: `identifyLayer()`, which identifies features within a specific layer and `identifyLayers()`, which identifies features for all layers in the current view.
```
Comment (hud10837) [inline kotlin/identify-layers/README.md]: Same as for java, these seem to be async only

**Comment by hud10837** [comment]
Sorry @sorenoid I've only just seen the comment rachael made about the method names in the API section, so feel free to disregard roughly 75% of those comments 😅

File: java/identify-layers/README.md (hunk 9 above)
Comment (sorenoid) [reply java/identify-layers/README.md]: thanks Hudson, this is a good catch!

File: java/integrated-windows-authentication/README.md [hunk 18]
```diff
@@ -1,32 +1,47 @@
...
-3. If the user authenticates, the search returns a list of web maps from `PortalItem` objects and the user can select one to display as an `ArcGISMap`.
+1. The `AuthenticationManager` object is configured with a challenge handler that will prompt for a Windows login (username, password, and domain) if a secure resource is encountered.
+2. When a search for portal items is performed against an IWA-secured portal, the challenge handler creates an `ArcGISNetworkCredential` object from the information entered by the user.
+3. If the user authenticates, the search returns a list of web maps (`ArcGISPortalItem`) and the user can select one to display as a `Map`.
+4. On some platforms, the current Windows account is used by default and a login prompt will not be shown if it can authenticate successfully.
 
 ## Relevant API
 
+* ArcGISNetworkCredential
+* ArcGISPortal
```
[the remark this answers is not in the document]
Comment (sorenoid) [reply java/integrated-windows-authentication/README.md]: sorry I thought this might be a parent class, I am still working on this massive API

**Comment by sorenoid** [comment]
just pinging @Rachael-E since since all have reviewed and I have resolved all conversations.

File: java/location-line-of-sight/README.md [hunk 19]
```diff
@@ -1,25 +1,29 @@
...
 
 ## How it works
-A `LineOfSight` analysis is a type of visual analysis you can perform on a scene. The `LineOfSight` analysis aims to answer the question: 'What are the visible and obstructed portions of a line between two locations?'. The output is a line, in an overlay, with two different colors - one representing visible areas, and the other representing obstructed areas.
 
-1. Create a `LocationLineOfSight`, and assign values for the `observerLocation`, and `targetLocation`.
-1. Once the `LocationLineOfSight` is created, add it to an `AnalysisOverlay`, and add the `AnalysisOverlay` to the `SceneView`.
-1. The first screen tap, sets the `observerLocation`.
-1. Each subsequent screen tap, sets the `targetLocation` and creates a new `LocationLineOfSight`.
+1. Create an `AnalysisOverlay` and add it to the scene view.
+1. Make a `DefaultSceneViewOnTouchHandler` to capture single taps.
```
Comment (Rachael-E) [inline java/location-line-of-sight/README.md]: Just need to change to the following and this looks good! 
```suggestion
1. Make a `DefaultSceneViewOnTouchListener` to capture single taps.
```

**Pull request merged**